src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -2403,7 +2403,7 @@ private static CodeType GetDefaultQueryParameterType()
         };
     }
     private static CodeType GetQueryParameterType(OpenApiSchema schema) =>
-        new()
+        GetPrimitiveType(schema) ?? new()
```

Comment:
@baywet, is it possible for `GetPrimitiveType` to return null?

Reply:
Yeah. This is possible. It ideally only returns a value when the schema is a primitive. 

So, it would ideally return null for a schema of type object or an empty schema like the ones below. 

```yaml
schema:
  type: "array"
  items: {}
```

```yaml
schema: {}
```